service/controller/cloudconfig/worker_template.go

```diff
@@ -1,51 +1,63 @@
 package cloudconfig
 
 import (
+	"context"
 	"fmt"
 
-	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
-	"github.com/giantswarm/certs"
-	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v7/pkg/template"
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/certs/v3/pkg/certs"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v8/pkg/template"
 	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/kvm-operator/service/controller/key"
 )
 
 // NewWorkerTemplate generates a new worker cloud config template and returns it
 // as a base64 encoded string.
-func (c *CloudConfig) NewWorkerTemplate(customObject v1alpha1.KVMConfig, data IgnitionTemplateData, node v1alpha1.ClusterNode, nodeIndex int) (string, error) {
-	var err error
-
-	var params k8scloudconfig.Params
+func (c *CloudConfig) NewWorkerTemplate(ctx context.Context, cr v1alpha1.KVMConfig, data IgnitionTemplateData, node v1alpha1.ClusterNode, nodeIndex int) (string, error) {
+	var extension *workerExtension
 	{
-		params = k8scloudconfig.DefaultParams()
-
-		params.BaseDomain = key.BaseDomain(customObject)
-		params.Cluster = customObject.Spec.Cluster
-		params.Extension = &workerExtension{
-			certs:        data.ClusterCerts,
-			customObject: customObject,
+		certFiles, err := fetchCertFiles(ctx, data.CertsSearcher, key.ClusterID(cr), workerCertFiles)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+		extension = &workerExtension{
+			certs:        certFiles,
+			customObject: cr,
 			nodeIndex:    nodeIndex,
 		}
+	}
+
+	var params k8scloudconfig.Params
+	{
+		params.BaseDomain = key.BaseDomain(cr)
+		params.Cluster = cr.Spec.Cluster
+		params.Extension = extension
 		params.Images = data.Images
 		params.Versions = data.Versions
 		params.Node = node
 		params.RegistryMirrors = c.registryMirrors
 		params.SSOPublicKey = c.ssoPublicKey
+		params.ImagePullProgressDeadline = "1m"
```

Comment:
We should put the value of this in some kind of key function. It does not make sense to me to have this magic number for both masters and workers.

Reply:
Agreed. I will fix this.